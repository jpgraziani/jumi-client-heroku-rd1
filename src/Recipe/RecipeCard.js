import React from 'react'
import { Link } from 'react-router-dom'
import { format } from 'date-fns'
import ApiContext from '../ApiContext'
import config from '../config'

export default class RecipeCard extends React.Component {
  static defaultProps ={
    onDeleteRecipe: () => {},
  }
  static contextType = ApiContext;

  handleClickDelete = e => {
    e.preventDefault()
    const recipeId = this.props.id

    fetch(`${config.API_ENDPOINT}/recipes/${recipeId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
          return res.json()
      })
      .then(() => {
        this.context.deleteRecipe(recipeId)
        this.props.onDeleteRecipe(recipeId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const { name, id, created, main_protein } = this.props
    return (
      <section>
          <div className='featured-recipes-center'>
            <Link to={`/recipe/${id}`}>
              <h3>{name} | {main_protein} | {format(created, 'Do MMM YYYY')}</h3>
            </Link>
          </div>
          <button
            className='btn-primary black-btn'
            type='button'
            onClick={this.handleClickDelete}
          >
            remove
          </button>
        </section>
    )
  }
}
