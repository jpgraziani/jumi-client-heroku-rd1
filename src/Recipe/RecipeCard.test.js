import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RecipeCard from './RecipeCard'

describe(`RecipeCard component`, () => {
  it('renders a .RecipeCard by default', () => {
    const wrapper = shallow(<RecipeCard />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})