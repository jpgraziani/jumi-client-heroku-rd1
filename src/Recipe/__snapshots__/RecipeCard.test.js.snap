// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard component renders a .RecipeCard by default 1`] = `
<section>
  <div
    className="featured-recipes-center"
  >
    <Link
      replace={false}
      to="/recipe/undefined"
    >
      <h3>
         | 
         | 
        Invalid Date
      </h3>
    </Link>
  </div>
  <button
    className="btn-primary black-btn"
    onClick={[Function]}
    type="button"
  >
    remove
  </button>
</section>
`;
